### PR TITLE
Adding minimum stake for create validator and delegation

### PR DESCRIPTION
**Pull request by gupadhyaya**
Restrictions imposed for create validator and delegation:
* create validator must have 10,000 ONE
* delegation must have at least 1000 ONE

### CODE REVIEW

File: staking/types/validator.go [hunk 1]
```diff
@@ -327,6 +336,12 @@ func (w *ValidatorWrapper) SanityCheck(
 				"have %s want %s", w.Delegations[0].Amount.String(), w.Validator.MinSelfDelegation,
 			)
 		}
+		if w.Delegations[0].Amount.Cmp(minimumStake) < 0 {
```
Comment (rlan35) [inline staking/types/validator.go]: just check the MinSelfDelegation is >= 10000ONE, it is checking against 1 ONE now. inside Validator.SanityCheck.

Comment (gupadhyaya) [reply staking/types/validator.go]: updated.

File: core/staking_verifier_test.go [hunk 2]
```diff
@@ -543,21 +545,21 @@ func TestCV32(t *testing.T) {
...
-	// MinSelfDelegation < 1 ONE
-	msg.MinSelfDelegation = big.NewInt(1e18 - 1)
+	statedb.AddBalance(msg.ValidatorAddress, tenK)
+	// MinSelfDelegation < 10,000 ONE
+	msg.MinSelfDelegation = big.NewInt(1e18)
 	if _, err := VerifyAndCreateValidatorFromMsg(
 		statedb, postStakingEpoch, big.NewInt(0), msg,
 	); err == nil {
-		t.Error("expected", "delegation-given 999999999999999999: min_self_delegation has to be greater than 1 ONE", "got", nil)
+		t.Error("expected", "delegation-given 1000000000000000000: min_self_delegation has to be greater than 10,000 ONE", "got", nil)
```
Comment (stephen-tse) [inline core/staking_verifier_test.go]: we can allow 10k OR greater? so that it's easier to create a validator at 10k self delegation, instead of 10k+1

Comment (stephen-tse) [reply core/staking_verifier_test.go]: @john-harmony we should update https://docs.harmony.one/home/validators/first-time-setup/creating-a-validator to have 10k+1? currently, 

```
--validator-addr [ONE ADDRESS] --amount 10 \
--min-self-delegation has to be at least 10,000 ONE.
```

**Pull request merged**